cmd/nsm-init/nsm-init.go

```diff
@@ -260,3 +268,49 @@ func getNetworkServices(nsmClient nsmconnect.ClientConnectionClient) ([]*netmesh
 		}
 	}
 }
+
+func requestConnection(nsmClient nsmconnect.ClientConnectionClient, cReq *nsmconnect.ConnectionRequest) (*nsmconnect.ConnectionParameters, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), clientConnectionTimeout)
+	defer cancel()
+	ticker := time.NewTicker(clientConnectionRetry)
+	defer ticker.Stop()
+	var err error
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, fmt.Errorf("nsm client: Connection request did not succeed within %d seconds, last known error: %+v", clientConnectionTimeout, err)
+		case <-ticker.C:
+			cResp, err := nsmClient.RequestConnection(ctx, cReq)
+			if err == nil && cResp.Accepted {
+				return cResp.ConnectionParameters, nil
+			}
+			logrus.Infof("nsm client: Connection request failed with: %+v, re-attempting in %d seconds",
+				err, clientConnectionRetry)
+			if cResp != nil {
+				logrus.Infof("nsm client: server side admission response: %s", cResp.AdmissionError)
+			}
+		}
+	}
+}
+
+func getCurrentNS() string {
+	for ln := 128; ; ln *= 2 {
```

Comment:
This being hardcoded to 128 scares me, is there a better way to the NS, and if not, is there some other maximum length value we could use? This will definitely bite us at some point if we leave it here.

Reply:
Agree, I checked the doc and Linux enforces the maximum length of symbolic link to 8k, so OS will not allow anything large than that. If I define a constant with 8k and with comment, does it work fo you?

Reply:
That works for me, thanks!

Reply:
done